alicloud/resource_alicloud_db_instance.go

```diff
@@ -785,7 +816,35 @@ func buildDBCreateRequest(d *schema.ResourceData, meta interface{}) (*rds.Create
 		request.ZoneId = Trim(zone.(string))
 	}
 
+	if zoneSlaves, ok := d.GetOk("zone_id_slaves"); ok {
+		for i, zone := range zoneSlaves.([]interface{}) {
+			*requestZoneIdSlaves[i] = Trim(zone.(string))
+		}
+	}
+
+	slaveVswitchesIds := []string{}
+	if vswitchIdSlaves, ok := d.GetOk("vswitch_id_slaves"); ok {
+		for _, slaveVswitchId := range vswitchIdSlaves.([]interface{}) {
+			slaveVswitchesIds = append(slaveVswitchesIds, Trim(slaveVswitchId.(string)))
+		}
+	}
+
 	vswitchId := Trim(d.Get("vswitch_id").(string))
+	// RDS API expects all vswitches to be a coma-separated list in VswitchID parameter
+	if len(slaveVswitchesIds) > 0 {
+		vswitchId = strings.Join(append([]string{vswitchId}, slaveVswitchesIds...), ",")
+	}
+
+	// Set zoneIdSlaves based on slave vswitches zones if zoneIDSlaves are not specified
+	for i, slaveVswitchId := range slaveVswitchesIds {
+		if *requestZoneIdSlaves[i] == "" {
+			vsw, err := vpcService.DescribeVSwitch(slaveVswitchId)
+			if err != nil {
+				return nil, WrapError(err)
+			}
+			*requestZoneIdSlaves[i] = vsw.ZoneId
+		}
+	}
 
```

Comment:
It seems like the `requestZoneIdSlaves` is never used in request. Please check it.

Reply:
Hi @xiaozhu36 

Thanks for reviewing this PR.

The parser is probably confused by the fact that `requestZoneIdSlaves` is a variable that hold reference to `request.ZoneIdSlave1` and `request.ZoneIdSlave2`, so that I can simply update these variables in loop iteration.
It doesn't have to be used after, it's just equivalent to do: `request.ZoneIdSlave1 = vsw.zoneId` and `request.ZoneIdSlave2 = vsw.zoneId`

Reply:
Hi @xiaozhu36 

Will you have some time to review that PR ?